tests/serde.rs

```diff
@@ -396,6 +396,47 @@ fn maps_objects_string_key() {
     test_via_json_with_config(src, Serializer::new().serialize_maps_as_objects(true));
 }
 
+#[wasm_bindgen_test]
+fn serde_deserialize_with() {
+    fn string_bool<'de, D>(de: D) -> Result<bool, D::Error>
+    where
+        D: serde::de::Deserializer<'de>,
+    {
+        struct StringBool {}
+
+        impl<'de> serde::de::Visitor<'de> for StringBool {
+            type Value = bool;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                write!(formatter, "a string")
+            }
+
+            fn visit_string<E>(self, v: String) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                Ok(v == "true")
+            }
+        }
+
+        de.deserialize_bool(StringBool {})
```

Comment:
I think the real problem here is that you're invoking `deserialize_bool`, even though you're actually expecting a string. Instead, you should give the deserializer a correct hint by invoking `deserialize_string`.

An even easier and correct way to implement this function would've been:

```rust
fn string_bool<'de, D>(de: D) -> Result<bool, D::Error>
where
  D: serde::de::Deserializer<'de>,
{
  String::deserialize(de).map(|v| v == "true")
}
```

Hence, I'm going to close this PR.